app/src/main/java/net/gaast/giggity/Schedule.java

```diff
@@ -355,7 +365,68 @@ private void loadXml(BufferedReader in, ContentHandler parser) {
 			throw new LoadException("XML parsing problem: " + e);
 		}
 	}
-	
+	private void loadJson(BufferedReader in) {
+
+		StringBuffer buffer=new StringBuffer();
+		SimpleDateFormat df=new SimpleDateFormat();
+		 HashMap<String,Schedule.Line> tentMap=new HashMap<String,Schedule.Line>();
+		Scanner s=new Scanner(in);
+		while (s.hasNext()){
+			buffer.append(s.nextLine());
+
+		}
+		String output=buffer.toString();
+		Schedule.Line line=null;
+		try {
+			JSONArray events=new JSONArray(output);
+			for(int i=0;i<events.length();i++){
+				JSONObject event=events.getJSONObject(i);
+				String uid=event.getString("id");
+				String title=event.getString("title");
+				String startTimeS=event.getString("start_time");
+				String endTimeS=event.getString("end_time");
+				Date startTime,endTime;
+				startTime = new Date(getTimeInMillis(startTimeS));
+				endTime = new Date(getTimeInMillis(endTimeS));
+				Schedule.Item item=new Schedule.Item(uid,title,startTime,endTime);
+				item.setDescription(event.getString("long_abstract"));
+				if(event.getString("signup_url")!="null"&&event.getString("signup_url")!=null){
+					item.addLink(new Link(event.getString("signup_url")));
+				}
+				JSONObject microlocation=event.getJSONObject("microlocation");
+				String location=microlocation.getString("name");
+				if ((line = tentMap.get(location)) == null) {
+					line = new Schedule.Line(location, location);
+					tents.add(line);
+					tentMap.put(location, line);
+				}
+				line.addItem(item);
+			}
+
+		} catch (JSONException e) {
+			e.printStackTrace();
+		}
+	}
+
+	private long getTimeInMillis(String timeString) {
+		GregorianCalendar calender=new GregorianCalendar();
+		String[] DateTime=new String[2];
+		String[]Date=new String[3];
+		String[]Time=new String[3];
+		DateTime=timeString.split("T");
+		Date=DateTime[0].split("-");
+		Time=DateTime[1].split(":");
+		int year=Integer.parseInt(Date[0]);
+		int month=Integer.parseInt(Date[1]);
+		int date=Integer.parseInt(Date[2]);
+		int hour=Integer.parseInt(Time[0]);
+		int min=Integer.parseInt(Time[1]);
+		int sec=Integer.parseInt(Time[2]);
+		calender.set(year,month,date,hour,min,sec);
+		return  calender.getTimeInMillis();
+	}
```

Comment:
Hrm, I *think* I had to do something quite similar in the Pentabarf parser already and IIRC it wasn't that complicated. Does Java not have a nicer way to do this?

(Also I saw OpenEvent has native timezone handling which is rather nice. Is that dealt with properly here? (Question is actually: what is properly :-))